display_list/display_list_image_filter.h

```diff
@@ -665,6 +672,70 @@ class DlColorFilterImageFilter final : public DlImageFilter {
   std::shared_ptr<DlColorFilter> color_filter_;
 };
 
+class DlLocalMatrixImageFilter final : public DlImageFilter {
+ public:
+  explicit DlLocalMatrixImageFilter(const SkMatrix& matrix,
+                                    std::shared_ptr<DlImageFilter> filter)
+      : matrix_(matrix), image_filter_(filter) {}
+  explicit DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter* filter)
+      : DlLocalMatrixImageFilter(filter->matrix_, filter->image_filter_) {}
+  DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter& filter)
+      : DlLocalMatrixImageFilter(&filter) {}
+
+  std::shared_ptr<DlImageFilter> shared() const override {
+    return std::make_shared<DlLocalMatrixImageFilter>(this);
+  }
+
+  DlImageFilterType type() const override {
+    return DlImageFilterType::kLocalMatrixFilter;
+  }
+  size_t size() const override { return sizeof(*this); }
+
+  const SkMatrix& matrix() const { return matrix_; }
+
+  const DlLocalMatrixImageFilter* asLocalMatrix() const override {
+    return this;
+  }
+
+  bool modifies_transparent_black() const override { return false; }
+
+  SkRect* map_local_bounds(const SkRect& input_bounds,
+                           SkRect& output_bounds) const override {
+    output_bounds = matrix_.mapRect(input_bounds);
+    return &output_bounds;
+  }
+
+  SkIRect* map_device_bounds(const SkIRect& input_bounds,
+                             const SkMatrix& ctm,
+                             SkIRect& output_bounds) const override {
+    return image_filter_->map_device_bounds(
+        input_bounds, SkMatrix::Concat(ctm, matrix_), output_bounds);
+  }
+
+  SkIRect* get_input_device_bounds(const SkIRect& output_bounds,
+                                   const SkMatrix& ctm,
+                                   SkIRect& input_bounds) const override {
+    return image_filter_->get_input_device_bounds(
```

Comment:
Same as `map_device_bounds` - which coordinate system should the returned bounds be relative to?